.github/workflows/ci.yaml

```diff
@@ -0,0 +1,56 @@
+name: CI
+
+on: pull_request
+
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ["2.7"]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set up Ruby environment
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - name: Run Rubocop
+        run: |
+          bundle exec rubocop
+
+  test:
+    name: Tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ["2.7"]
+        puppet: ["7"]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set up Puppet environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y puppet-agent=${{ matrix.puppet }}
+
```

Comment:
Please use `PUPPET_VERSION`, see https://github.com/theforeman/kafo/blob/4d3440c02a6592a3de34e8205688a118aa5cf360/Gemfile#L8

Reply:
And you can set it like this: https://github.com/voxpupuli/beaker-hostgenerator/blob/61c7244c239c5e23a85ea3acb4952ff9d0ea9630/.github/workflows/test.yml#L35-L37